cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/ConfigurationFinderTest.java

```diff
@@ -83,4 +97,210 @@ public void invalidPathTest() {
+    Path propertieFile = Files.createFile(userHome.toPath().resolve(".cobigen"));
+    List<String> properties = Arrays.asList("templates=" + customTemplatesFolder.getAbsolutePath());
+    Files.write(propertieFile, properties);
+
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI templates = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(templates)).exists();
+      assertThat(templates).isEqualTo(customTemplatesFolder.toURI());
+    });
+  }
+
+  /**
+   * Test of findTemplates with a template-set value set in the cobigen file, the value is faulty and the folder does
+   * not exist. Expecting to create a template-set folder in the COBIGEN_HOME home directory.
+   */
+  @Test
+  public void findTemplateSetWithInvalidProperties() throws Exception {
+
+    File userHome = this.tmpFolder.newFolder("user-home");
+    File customTemplatesFolder = this.tmpFolder.newFolder("customTemplatesFolder");
+    Path propertieFile = Files.createFile(userHome.toPath().resolve(".cobigen"));
+    List<String> properties = Arrays
+        .asList("template-sets=" + customTemplatesFolder.getAbsolutePath().replaceFirst("custom", "nonExistend"));
+    Files.write(propertieFile, properties);
+
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI templates = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(templates)).exists();
+      assertThat(Paths.get(templates))
+          .isEqualTo(userHome.toPath().resolve(ConfigurationConstants.TEMPLATE_SETS_FOLDER));
+    });
+  }
+
+  /**
+   * Test of findTemplates with a template-set value set in the cobigen file, expecting to return the value from the
+   * .cobigen file
+   */
+  @Test
+  public void findTemplateSetWithProperties() throws Exception {
+
+    File userHome = this.tmpFolder.newFolder("user-home");
+    File customTemplatesFolder = this.tmpFolder.newFolder("customTemplatesFolder",
+        ConfigurationConstants.TEMPLATE_SETS_FOLDER);
+    Path propertieFile = Files.createFile(userHome.toPath().resolve(".cobigen"));
+    List<String> properties = Arrays.asList("template-sets=" + customTemplatesFolder.getAbsolutePath());
+    Files.write(propertieFile, properties);
+
+    withEnvironmentVariable(ConfigurationConstants.CONFIG_ENV_HOME, userHome.getAbsolutePath()).execute(() -> {
+      URI templateSets = ConfigurationFinder.findTemplatesLocation();
+      assertThat(Paths.get(templateSets)).exists();
+      assertThat(templateSets).isEqualTo(customTemplatesFolder.toURI());
+    });
+  }
+
+  /**
+   *
+   * Test of findTemplates with a templates value set in the cobigen file, the value is faulty and the folder does not
+   * exist. Expecting to create a template-set folder in the COBIGEN_HOME home directory and return the folder.
+   */
+  @Test
+  public void findTemplatesWithInvalidProperties() throws Exception {
```

Comment:
Maybe rename to findTemplatesWithInvalidPropertiesCreatesFolder?

Reply:
Yeah, I don't know if we should put every detail into the method name. The description of the tests already says that a folder should be created, I just picked the names to be simple and make the test quickly distinguishable.